src/snarky.d.ts

```diff
@@ -18,31 +18,124 @@ export {
+ * `Provable<T>` is the required input type in a few places in SnarkyJS. One convenient way to create a `Provable<T>` is using `Struct`.
  * 
  * The properties and methods on the provable type exist in all base SnarkyJS types as well (aka. {@link Field}, {@link Bool}, etc.). In most cases, a zkApp developer does not need these functions to create Dapps.
  */
 declare interface Provable<T> {
-  toFields: (x: T) => Field[];
-  toAuxiliary: (x?: T) => any[];
-  fromFields: (x: Field[], aux: any[]) => T;
+  /**
+   * A function that takes `value`, an element of type `T`, as argument and returns an array of {@link Field} elements that made up the provable data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the {@link Field} array from.
+   * 
+   * @return A {@link Field} array describing how this `T` element is made up of {@link Field} elements.
+   */
+  toFields: (value: T) => Field[];
+
+   /**
+   * A function that takes `value` (optional), an element of type `T`, as argument and returns an array of any type that made up the "auxilary" (non-provable) data of `value`.
+   * 
+   * @param value - the element of type `T` to generate the auxilary data array from, optional. If not provided, an empty array is returned.
+   * 
+   * @return An array of any type describing how this `T` element is made up of "auxilary" (non-provable) data.
+   */
+  toAuxiliary: (value?: T) => any[];
+
+  /**
+   * A function that returns an element of type `T` from the given provable and "auxilary" data.
+   * 
+   * **Important**: For any element of type `T`, this function is the reverse operation of calling {@link toFields} and {@link toAuxilary} methods on an element of type `T`.
+   * 
+   * @param fields - an array of {@link Field} elements describing the provable data of the new `T` element.
+   * @param aux - an array of any type describing the "auxilary" data of the new `T` element, optional.
+   * 
+   * @return An element of type `T` generated from the given provable and "auxilary" data.
+   */
+  fromFields: (fields: Field[], aux: any[]) => T;
+
+  /**
+   * Return the size of the `T` type in terms of {@link Field} type, as {@link Field} is the primitive type.
+   * 
+   * **Warning**: This function returns a `number`, so you cannot use it to prove something on chain. You can use it during debugging or to understand the memory complexity of some type.
+   * 
+   * **Beware**: Calling the statement `x.sizeInFields()` is equivalent to `x.toFields().length`.
+   * 
+   * @return A `number` representing the size of the `T` type in terms of {@link Field} type.
+   */
   sizeInFields(): number;
-  check: (x: T) => void;
+
+  /**
+   * Assert some statements on the chain to check if `value` element is a valid member of type `T`.
+   * This function does not return anything, rather it creates any number of assertions on the chain to prove `value` is a valid member of the type `T`.
```

Comment:
I don't like the terminology "on the chain" to refer to constraints added to the proof. The proof is created locally, and not even necessarily sent to a blockchain (zk proofs are a generic tool, they could be used on Mina, but also in other contexts).

How about this:
```suggestion
   * Add assertions to the proof to check if `value` is a valid member of type `T`.
   * This function does not return anything, instead it creates any number of assertions to prove that `value` is a valid member of the type `T`.
```